cpp/include/raft/stats/accuracy.cuh

```diff
@@ -39,6 +40,29 @@ float accuracy(const math_t* predictions, const math_t* ref_predictions, int n,
   return detail::accuracy_score(predictions, ref_predictions, n, stream);
 }
 
+/**
+ * @brief Compute accuracy of predictions. Useful for classification.
+ * @tparam math_t: data type for predictions (e.g., int for classification)
+ * @tparam IdxType Index type of matrix extent.
+ * @tparam LayoutPolicy Layout type of the input data.
+ * @tparam AccessorPolicy Accessor for the input and output, must be valid accessor on
+ *                        device.
+ * @param[in] handle: the raft handle.
+ * @param[in] predictions: array of predictions (GPU pointer).
+ * @param[in] ref_predictions: array of reference (ground-truth) predictions (GPU pointer).
+ * @return: Accuracy score in [0, 1]; higher is better.
+ */
+template <typename math_t, typename IdxType, typename LayoutPolicy, typename AccessorPolicy>
+float accuracy(
+  const raft::handle_t& handle,
+  raft::mdspan<math_t, raft::vector_extent<IdxType>, LayoutPolicy, AccessorPolicy> predictions,
+  raft::mdspan<math_t, raft::vector_extent<IdxType>, LayoutPolicy, AccessorPolicy> ref_predictions)
+{
+  return detail::accuracy_score(predictions.data_handle(),
```

Comment:
Would you consider first `static_assert`ing that the layouts are always exhaustive, strided, and unique, before handing them off to a function that assumes these properties?

(This suggestion generally applies to all the functions in this PR.)

Reply:
Thanks!  I should perhaps have also written that if we're calling into some existing raw array code, then we're probably baking in the assumption of layout left or layout right.

It's OK to use `.size()` here as it's the product of the one extent, but normally I would spell this `.extent(0)` instead of `.size()`.